src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -635,20 +631,39 @@ private void SerializeAsV2(
 
         private void SerializeTypeProperty(JsonSchemaType? type, IOpenApiWriter writer, OpenApiSpecVersion version)
         {
+            // check whether nullable is true for upcasting purposes
+            var isNullable = Nullable ||
+                            Type is JsonSchemaType.Null ||
```

Comment:
```suggestion
                            (Type.HasValue && Type.Value.HasFlag(JsonSchemaType.Null))  ||
```
If I'm not wrong, the use of `is` won't return true if the enum has more than one flag set. e.g when Type is set as `JsonSchemaType.Object | JsonSchemaType.Null`

Is this the intention here?

Reply:
thanks! great catch here!